app/src/main/java/com/mr/nemo/dragonfly/ui/screen/auth/signup/SignUpScreen.kt

```diff
@@ -1,58 +1,155 @@
+) {
+    val state by viewModel.state.collectAsStateWithLifecycle()
+
+    SignUpScreen(
+        state = state,
+        onEvent = viewModel::onEvent
+    )
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
+@Composable
+fun SignUpScreen(
+    state: SignUpScreenState,
+    onEvent: (SignUpScreenEvent) -> Unit,
     modifier: Modifier = Modifier
 ) {
     val spacing = DragonFlyTheme.spacing
     val typography = DragonFlyTheme.typography
     val colors = DragonFlyTheme.colors
+
+    val page = state.pages[state.currentPage]
+
     Scaffold(
+        modifier = modifier,
         topBar = {
-            TopAppBar(
-                title = { /*TODO*/ },
-                navigationIcon = {
-                    Icon(
-                        painter = painterResource(id = R.drawable.ic_arrow_back),
-                        contentDescription = stringResource(id = R.string.content_description_navigate_back),
-                        modifier = Modifier.clickable { }
-                    )
-                },
-                actions = {
-                    Icon(
-                        painter = painterResource(id = R.drawable.ic_info),
-                        contentDescription = stringResource(id = R.string.content_description_sign_up_faq),
-                        modifier = Modifier.clickable { }
+            Column {
+                TitledTopAppBar(
+                    title = page.title,
+                    navigationIcon = {
+                        IconButton(onClick = { onEvent(SignUpScreenEvent.OnBackClicked) }) {
+                            Icon(
+                                painter = painterResource(id = R.drawable.ic_arrow_back),
+                                contentDescription = stringResource(id = R.string.content_description_navigate_back)
+                            )
+                        }
+                    },
+                    actions = {
+                        IconButton(onClick = { onEvent(SignUpScreenEvent.OnInfoClicked) }) {
+                            Icon(
+                                painter = painterResource(id = R.drawable.ic_info),
+                                contentDescription = stringResource(id = R.string.content_description_sign_up_faq),
+                                tint = Color.Unspecified
+                            )
+                        }
+                    },
+                    modifier = Modifier.padding(horizontal = spacing.medium)
+                )
+                AnimatedVisibility(visible = page.isProgressVisible) {
+                    LinearProgressIndicator(
+                        progress = state.progress,
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .requiredHeight(8.dp),
+                        color = colors.primary.main,
+                        trackColor = colors.neutral7,
+                        strokeCap = StrokeCap.Square
```

Comment:
@bohdanzdaniekin as a reminder, we already discussed that the progress bar should be part of the main content and not related to the top bar